Utilities/rANS/include/rANS/Encoder.h

```diff
@@ -122,22 +146,41 @@ const stream_IT Encoder<coder_T, stream_T, source_T>::Encoder::process(
                             mProbabilityBits);
     ransCoder::encPutSymbol(&rans0, &ptr, &(*mSymbolTable)[s0],
                             mProbabilityBits);
+    assert(&(*outputBegin) < ptr);
   }
   ransCoder::encFlush(&rans1, &ptr);
   ransCoder::encFlush(&rans0, &ptr);
 
-  try {                            //TODO Michael may want to generate exception message in different way
-    assert(&(*outputBegin) < ptr); // for some reason assert does not work in test, apparently BOOST modifies its handling
-    if (ptr < &(*outputBegin)) {   // RS: this exception is thrown with default calculateMaxBufferSize when running o2-test-ctf-io
-      throw std::runtime_error(o2::utils::concat_string("output buffer too short: provided ",
-                                                        std::to_string(&(*outputEnd) - &(*outputBegin)),
-                                                        " filled ", std::to_string(&(*outputEnd) - ptr), " slots"));
-    }
-  } catch (std::exception& e) {
-    std::cerr << "Exception is thrown: " << e.what() << '\n';
-    throw;
+  assert(&(*outputBegin) < ptr);
+
+  // deal with overflow
+  if (ptr < &(*outputBegin)) {
+    const std::string exceptionText = [&]() {
+      std::stringstream ss;
+      ss << __func__ << " detected overflow in encode buffer: allocated:" << std::distance(inputBegin, inputEnd) << ", used:" << std::distance(ptr, &(*outputEnd));
+      return ss.str();
+    }();
+
+    LOG(error) << exceptionText;
+    throw std::runtime_error(exceptionText);
   }
 
+  t.stop();
+  LOG(debug) << __func__ << " inclusive time (ms): " << t.getDurationMS();
+
+// advanced diagnostics for debug builds
+#if !defined(NDEBUG)
+  LOG(info) << "EncoderProperties: {"
+            << "sourceTypeB: " << sizeof(source_T) << ", "
+            << "streamTypeB: " << sizeof(stream_T) << ", "
+            << "coderTypeB: " << sizeof(coder_T) << ", "
+            << "probabilityBits: " << mProbabilityBits << ", "
+            << "inputBufferSizeB: " << inputBufferSize * sizeof(source_T) << ", "
+            << "outputBufferSizeB: " << std::distance(ptr, &(*outputEnd)) * sizeof(stream_T) << "}";
+#endif
```

Comment:
Why don't you just use LOG(DEBUG) in that case?
People might compile without -DNDEBUG, but still have the log verbosity to >= info

Reply:
Will take care of this, thanks for the comment.